test/integration/testLargeResultSet.js

```diff
@@ -113,4 +113,126 @@ describe('Large result Set Tests', function ()
       }
     });
   });
+
+  it('testVariantLarge', function (done)
+  {
+    const createTableWithVariant = 'create or replace table testVariantTable(colA variant)';
+    const insertVariant = 'insert into testVariantTable select value from table(flatten(parse_json(?)))';
+    const selectVariant = 'select * from testVariantTable';
+    const dropTableWithVariant = 'drop table if exists testVariantTable';
+
+    var arrJSON = [];
+    const sampleJSON = {
+      "root":
+      {
+        "key":
+          [
+            {
+              "key1": "value1",
+              "key2": "value2",
+              "key3": "value3",
+              "key4": "value4",
+              "key5":
+              {
+                "key":
+                  [
+                    { "key1": "value1", "key2": "value2" },
+                    { "key1": "value1", "key2": "value2" },
+                    { "key1": "value1", "key2": "value2" },
+                    { "key1": "value1", "key2": "value2" }
+                  ]
+              },
+              "key6":
+                [
+                  { "key1": "value1", "key": "value" },
+                  { "key1": "value1", "key": "value" },
+                  { "key1": "value1", "key": "value" },
+                  { "key1": "value1", "key": "value" },
+                  { "key1": "value1", "key": "value" },
+                  { "key1": "value1", "key": "value" },
+                  { "key1": "value1", "key": "value" }
+                ]
+            },
+          ]
+      }
+    };
+
+    for (var i = 0; i < sourceRowCount; i++)
+    {
+      arrJSON.push(sampleJSON);
+    }
+
+    async.series([
+      function (callback)
+      {
+        testUtil.executeCmd(connection, createTableWithVariant, callback);
+      },
+      function (callback)
+      {
+        connection.execute({
+          sqlText: insertVariant,
+          binds: [JSON.stringify(arrJSON)],
+          complete: function (err, stmt)
+          {
+            if (err)
+            {
+              callback(err);
+            }
+
+            try
+            {
+              assert.strictEqual(stmt.getNumUpdatedRows(), sourceRowCount);
```

Comment:
if this assertion fails then callback in line below won't be called and the test could stuck till timeout?